psi4/driver/driver_nbody.py

```diff
@@ -25,830 +25,1290 @@
+
+--------------------------
+ManyBodyComputer.compute()
+--------------------------
+* compute() for each job in self.task_list
+
+----------------------------------
+ManyBodyComputer.get_psi_results()
+----------------------------------
+
+    Computer.get_results()
+    ----------------------
+
+        Computer._prepare_results()
+        ---------------------------
+        * if multiple modelchems (multilevel):
+
+            multilevel.prepare_results()
+            ----------------------------
+            * separates out each modelchem calcs and calls _prepare_results on each
+            * sums modelchem levels and returns small dict back to get_results()
+
+        * call get_results() for each job in task list
+        * assemble all the computed energies, all the computed gradients, and all the computed hessians
+        * for each available derivative, call:
+
+            assemble_nbody_components()
+            ---------------------------
+            * re-call build_nbody_compute_list to get the cp/nocp/vmfc lists again
+
+                build_nbody_compute_list()
+                --------------------------
+
+            * slice up the supersystem mol into fragment atom ranges to help g/h arrays build piecemeal
+            * prepare empty {bsse_type}_by_level and {bsse_type}_body_dict structs. the former have different contents for vmfc
+            * for cp and nocp, resort the build_nbody_compute_list returns into per-body lists suitable for summing
+            * note that nb loops often run over more than active nbodies_per_mc_level item due to 1-body for subtraction and multilevel complications
+            * for each possibly active n-body level and each active bsse_type, call _sum_cluster_ptype_data to build by_level structs
+
+                _sum_cluster_ptype_data()
+                -------------------------
+                * build up ene, grad, Hess in per-fragment pieces based on list of (frag, bas) subjobs active for that bsse treatment
+
+            * compute special case of monomers in monomer basis
+            * for each of cp/nocp/vmfc, apply appropriate formula to build each n-body level of cummulative total energy into body_dict
+            * for driver=energy, set several qcvars and call:
+
+                _print_nbody_energy()
+                ---------------------
+                * prints and logs formatted energy output. called separately for cp, nocp, vmfc
+
+            * collect qcvars and summed levels into a return dictionary with some extra aliases for target bsse_type and target driver
+
+        * merge all the assemble_nbody_components return dictionaries
+        * in struct["intermediates"], store dict of `"N-BODY (?)@(?) TOTAL ENERGY" = return_energy` for all in task_list or results kwarg
+        * in struct["intermediates_{ptype}"], store dict of `task_list key = return_{ptype}` for all in task_list or results kwarg
```

Comment:
What kind of object is `return_{ptype}`? Are these E/G/H?

Reply:
yes, "ptype" was predecessor to "driver" and is still used for e/g/h in more local circumstances (e.g., driver=gradient often deals with ptype=energy and ptype=gradient)